Задача 1
Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Ckn=n!/k!(n−k)!


In [12]:
import numpy as np
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [13]:
C1 = combinations(4,1) * combinations(48,3)
C2 = combinations(4,2) * combinations(48,2)
C3 = combinations(4,3) * combinations(48,1)
C4 = combinations(4,4) * combinations(48,0)
C = C1 +C2+C3+C4
print (C)

76145


Ответ:76145

Задача 2
Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?


Общее число исходов 7!
P(A) = 12*5!/7!
P(B) = 5*6*4!/7!

In [16]:
P1 = 12 * factorial(5) / factorial(7)
print(P1)

0.2857142857142857


In [17]:
P2 = 5*6* factorial(4) / factorial(7)
print(P2)

0.14285714285714285


Задача 3
Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [19]:
C1 = combinations(50,3)
print(C1)
C2 = combinations(50,2)*combinations(10,1)
print(C2)
C3 = combinations(60,3)

19600
12250


In [21]:
P1 = C1 / C3
print(P1)
P2 = C2 / C3
print(P2)

0.5727644652250146
0.35797779076563413



Задача 4
Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

События являются независимыми, если P(A|B) = P(A)

P1 = P(A)
P2 = P(A|B)


In [23]:
P1 = 1/2
P2 = 2/3
print (P1)
print (P2)

0.5
0.6666666666666666


In [ ]:
Ответ: События А и B зависимы

Задача 5
Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.
Вопрос: какова вероятность, что вы действительно больны ей?
Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

P(A)=P(B1)⋅P(A|B1)+P(B2)⋅P(A|B2)+...+P(Bn)⋅P(A|Bn)
А - человек болен
B - Тест положительный
формула Байеса 
Вероятность, что ч. болен при положительном тесте
P(A|B) = P(A)* P(B|A)/P(B)

P1 = P(A) = 0.001
Вероятность положительного теста, если чел. болен
P2 = P(B|A) = 0.99
P3 = P(A2) -  вероятность что чел. не болен = 0,999
P4 = P(B|A2) - Вероятность ложноположит. теста 0.01
P5 = P(B)=P(A)⋅P(B|A)+P(A2)⋅P(В|A2)


In [24]:
P1 = 0.001
P2 = 0.99
P3 = 0.999
P4 = 0.01
P5 = P1*P2 + P3*P4
P = P1*P2/P5
print(P)

0.09016393442622951
